pkg/apis/controlplane/helper_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplane
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestGroupReferenceToString(t *testing.T) {
+	tests := []struct {
+		name       string
+		inGroupRef *GroupReference
+		out        string
+	}{
+		{
+			name: "cg-ref",
+			inGroupRef: &GroupReference{
+				Namespace: "",
+				Name:      "cgA",
+			},
+			out: "cgA",
+		},
+		{
+			name: "g-ref",
+			inGroupRef: &GroupReference{
+				Namespace: "nsA",
+				Name:      "gA",
+			},
+			out: "nsA/gA",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			actualOut := tt.inGroupRef.ToString()
+			assert.Equal(t, tt.out, actualOut)
+		})
+	}
+}
+
+func TestGroupReferenceToTypedString(t *testing.T) {
+	tests := []struct {
+		name       string
+		inGroupRef *GroupReference
+		out        string
+	}{
+		{
+			name: "cg-ref",
+			inGroupRef: &GroupReference{
+				Namespace: "",
+				Name:      "cgA",
+			},
+			out: "ClusterGroup: cgA",
+		},
+		{
+			name: "g-ref",
+			inGroupRef: &GroupReference{
+				Namespace: "nsA",
+				Name:      "gA",
+			},
+			out: "Group: nsA/gA",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			actualOut := tt.inGroupRef.ToTypedString()
+			assert.Equal(t, tt.out, actualOut)
+		})
+	}
+}
+
+func TestNetworkPolicyReferenceToString(t *testing.T) {
+	tests := []struct {
+		name    string
+		inNPRef *NetworkPolicyReference
+		out     string
+	}{
+		{
+			name: "acnp-ref",
+			inNPRef: &NetworkPolicyReference{
+				Type:      AntreaClusterNetworkPolicy,
+				Namespace: "",
+				Name:      "acnpA",
+			},
+			out: "AntreaClusterNetworkPolicy:acnpA",
```

Comment:
Just curious: why do we have a space between resource type and name for cg/group ToString(), but not for ANP?

Reply:
you mean the difference in NP.ToString() and Group/CG.ToTypedString() ?
The ToTypedString for groups is introduced for logging and other purposes. However the ToString for both objects is the same( no whitespace)

Reply:
I asked the same question as @Dyanngg above